src/qutip_qip/operations/gateclass.py

```diff
@@ -153,17 +158,45 @@ def __init__(
         classical_controls=None,
         classical_control_value: Optional[int] = None,
         arg_label=None,
+        style=None,
         **kwargs,
     ):
         """
         Create a gate with specified parameters.
         """
-
         self.name = name if name is not None else self.__class__.__name__
         self.targets = None
         self.controls = None
         self.classical_controls = None
 
+        if style is not None:
+            self.text = style.get("text", self.name)
+            self.color = style.get(
+                "color",
+                (
+                    default_theme[self.name]
+                    if self.name in default_theme
+                    else "#000000"
+                ),
+            )
+            self.fontsize = style.get("fontsize", 10)
+            self.fontcolor = style.get("fontcolor", "#FFFFFF")
+            self.fontweight = style.get("fontweight", "normal")
+            self.fontstyle = style.get("fontstyle", "normal")
```

Comment:
It is best to move this part also to the plotting files, not in the gate files. The default type could be simply `None`, and all the default settings will be handled when plotting the circuit.

This avoids some complicated cross-importation. `default_theme` only needs to be imported in the plotting code.

Reply:
Agreed, we can move it to `circuit.py` once the plotting API is finalized. For now, I've temporarily moved it to `mat_renderer.py`.

Reply:
Actually `mat_rendere.py` is good. We should avoid adding things to `circuit.py`. In fact, I hope that we can move the full plotting functions out of `circuit.py`.